### DISCUSSION TITLE
Problem with getting data out of an array and display it on dynamic page

### DISCUSSION THREAD

**Discussion by MeenaFoerderer** [neuefische/questions#383] [Web] [answered]
Hello, 
I have dynamic pages that display lists of items (conference talks). The data is an array in my helpers folder. Each object in the array is a conference talk that also has a property for "session". I want to display the sessions separately on the page on top of the whole list of rendered talks. However, I don't know how to do that.
I started with this code in the [date]/[room]/index.js (lines 88-90): 
`{filteredTalks.map((talk) => (<p>{talk.session}</p>))} `
That gives me a long list of duplicates of course. I don't know how to "reduce" this list.
It probably works with `Array.from(new Set(..)) ` but I'm not sure. 

My current PR: https://github.com/MeenaFoerderer/Capstone-project/pull/20

**Comment by ahohnsen** [comment, marked as answer]
Hey Meena,

First you need to `map()` over your `filteredTalks` and return `talk.session` for every talk. You will get an array (`allSessions`) with all sessions mentioned in your data set. As you already noticed, the session names will appear multiple times. To get a list without duplicates you can use `new Set` and spread that into an array. So in the end, you receive an array with every session name mentioned only once (`uniqueSessions`).

```jsx
  const allSessions = filteredTalks.map((talk) => talk.session);
  const uniqueSessions = [...new Set(allSessions)];
```

Hope, that helps :)
